[E2E-SIM] 1 failures — 2026-05-03T13:52:34.940Z

## E2E Simulation Failure Report

**Run ID**: `85350a7a-3cdf-421e-98e3-91fd0d20db71`
**Failed Checks**: 1/1
**Environment**: dev

## Failed Assertions
- **user_refund:user[d4fcf8dc-32da-4e30-af1c-591c61ec158d] refunds application[42453d0d-429e-4d32-9961-7b4f157348e6] for event[8d2590e7-fa0e-40e4-b90f-7cf60ca427f7]**: expected `"success"`, got `"expected status 200, got 400"`

## Axiom 디버깅
Edge Function 로그는 Axiom에서 확인할 수 있습니다.

```
axiom query "['edge-functions'] | where metadata.runId == '85350a7a-3cdf-421e-98e3-91fd0d20db71'"
```

또는 Axiom 대시보드에서 dataset `edge-functions`를 선택하고 `metadata.runId == '85350a7a-3cdf-421e-98e3-91fd0d20db71'`로 필터링하세요.

## Log Details
No storage log available

<details>
<summary>📋 Full Log (truncated to 60KB)</summary>

```log
[2026-05-03T13:52:06.418Z] [INFO] [tick:init] Found 5 E2E partners
[2026-05-03T13:52:07.406Z] [INFO] [tick:init] Selected 10/60 users for this tick
[2026-05-03T13:52:09.496Z] [WARN] [tick:partner_auth] No email found for partner 54435848-e269-44e3-9790-67309045246c
[2026-05-03T13:52:10.256Z] [WARN] [tick:partner_auth] No email found for partner 55032dc6-3aa5-4ef3-8c6b-0f9e964cd9d8
[2026-05-03T13:52:10.643Z] [WARN] [tick:partner_auth] No email found for partner c518ec3d-0988-4b47-9abc-6655763a45bb
[2026-05-03T13:52:11.352Z] [WARN] [tick:partner_auth] No email found for partner 2dae98b5-9f92-467b-9ad1-a590442f9cb6
[2026-05-03T13:52:11.639Z] [WARN] [tick:partner_auth] No email found for partner ae0f3970-6045-4a90-a9c0-5804fde1e2e1
[2026-05-03T13:52:32.181Z] [INFO] [tick:generate] Generated 1 actions
[2026-05-03T13:52:33.951Z] [ERROR] [tick:user_refund] user[d4fcf8dc-32da-4e30-af1c-591c61ec158d] refunds application[42453d0d-429e-4d32-9961-7b4f157348e6] for event[8d2590e7-fa0e-40e4-b90f-7cf60ca427f7] → ❌
{"efAssertion":{"passed":false,"name":"refund_ef_status","details":"expected status 200, got 400"},"dbAssertion":{"passed":false,"name":"refund_db_application_status","details":"expected status='cancelled', got 'approved'"}}
[2026-05-03T13:52:33.951Z] [ERROR] [tick:summary] Tick complete: 0/1 passed, 1 failed
```

</details>

--- a/supabase/BLUEDOC.md
+++ b/supabase/BLUEDOC.md
@@ -33,4 +33,4 @@ Minglit 의 **백엔드 구현 루트**. DB migration, Edge Functions, seed, bac
 - [docs/infra/bluedoc/BLUEDOC.md](../docs/infra/bluedoc/BLUEDOC.md) — BLUEDOC 컨벤션
 
 ---
-_Reviewed: 2026-06-09 13:54_
+_Reviewed: 2026-06-10 04:35_

--- a/supabase/functions/event-flow-simulator/action/new_actions_test.ts
+++ b/supabase/functions/event-flow-simulator/action/new_actions_test.ts
@@ -12,9 +12,46 @@ import { blockAction } from "./block.ts";
 import { partnerApproveAction } from "./partner_approve.ts";
 import { partnerRejectAction } from "./partner_reject.ts";
 import { partnerCreateEventAction } from "./partner_create_event.ts";
-import { createPRNG } from "../core/types.ts";
+import { createPRNG, type PRNG } from "../core/types.ts";
 
 const rng = createPRNG(1);
+const DAY_MS = 86_400_000;
+
+function sequenceRng(values: number[]): PRNG {
+  let index = 0;
+  return {
+    next() {
+      return values[index++] ?? values.at(-1) ?? 0;
+    },
+    split() {
+      return sequenceRng(values.slice(index));
+    },
+  };
+}
+
+function withFixedNow<T>(isoNow: string, fn: () => T): T {
+  const fixedNow = new Date(isoNow).getTime();
+  const originalNow = Date.now;
+  Date.now = () => fixedNow;
+  try {
+    return fn();
+  } finally {
+    Date.now = originalNow;
+  }
+}
+
+function createEventStartDelayDays(rng: PRNG): number {
+  const fixedNow = new Date("2026-06-08T00:00:00.000Z").getTime();
+  return withFixedNow("2026-06-08T00:00:00.000Z", () => {
+    const payload = partnerCreateEventAction.buildPayload({
+      myParties: [{ id: "party1", status: "active" }],
+      myEvents: [],
+    }, rng);
+    const startTime = (payload.event as Record<string, unknown>)
+      .start_time as string;
+    return (new Date(startTime).getTime() - fixedNow) / DAY_MS;
+  });
+}
 
 // ============================================================
 // Identity 가드 — type / role / ef wiring
@@ -241,6 +278,17 @@ Deno.test({
       Error,
       "without an existing party",
     );
+  },
+});
+
+Deno.test({
+  name:
+    "partnerCreateEventAction.buildPayload - start horizon crosses refund 7-day cutoff",
+  fn: () => {
+    // First rng value selects the party; second value selects start delay.
+    assertEquals(createEventStartDelayDays(sequenceRng([0, 0])), 2);
+    assertEquals(createEventStartDelayDays(sequenceRng([0, 0.39])), 7);
+    assertEquals(createEventStartDelayDays(sequenceRng([0, 0.999999])), 14);
   },
 });
 

--- a/supabase/functions/event-flow-simulator/action/partner_create_event.ts
+++ b/supabase/functions/event-flow-simulator/action/partner_create_event.ts
@@ -5,8 +5,15 @@ import { registerAction } from "./_registry.ts";
 
 const MIN_SCHEDULED_EVENTS = 2;
 const EVENT_DURATION_MS = 2 * 60 * 60 * 1000;
-const MIN_START_DELAY_DAYS = 7;
-const START_DELAY_WINDOW_DAYS = 21;
+// horizon 을 7일을 가로지르는 2~14일로 둔다.
+// - 과거 7~28일(중앙값 17일) 은 scheduled 가 수 주간 유지되어 모든 적격 파트너가
+//   MIN_SCHEDULED_EVENTS 캡에 묶이고 이벤트 공급이 영구 정지됐다 (auto_complete_past_events
+//   15분 cron 이 end_time 을 지나야 completed 로 내려 캡을 푼다).
+// - 7일 미만 이벤트는 며칠 안에 완료되어 캡을 풀고 create 를 재개시킨다.
+// - 7일 이상 이벤트(약 절반)는 남겨, refund 액션(event start ≥ 7일, Fix #2131)의
+//   적격성을 유지한다. 하한을 1일이 아닌 2일로 두는 것도 같은 이유.
+const MIN_START_DELAY_DAYS = 2;
+const START_DELAY_WINDOW_DAYS = 12;
 
 type PartnerParty = {
   id: string;

--- a/supabase/functions/event-flow-simulator/action/partner_create_event.ts
+++ b/supabase/functions/event-flow-simulator/action/partner_create_event.ts
@@ -13,7 +13,7 @@ const EVENT_DURATION_MS = 2 * 60 * 60 * 1000;
 // - 7일 이상 이벤트(약 절반)는 남겨, refund 액션(event start ≥ 7일, Fix #2131)의
 //   적격성을 유지한다. 하한을 1일이 아닌 2일로 두는 것도 같은 이유.
 const MIN_START_DELAY_DAYS = 2;
-const START_DELAY_WINDOW_DAYS = 12;
+const START_DELAY_WINDOW_DAYS = 13;
 
 type PartnerParty = {
   id: string;

--- a/supabase/seed.dev.sql
+++ b/supabase/seed.dev.sql
@@ -793,3 +793,35 @@ BEGIN
     WHERE id = existing_id AND is_verified IS DISTINCT FROM true;
   END IF;
 END $$;
+
+-- ── Phase 11: Simulator partner party backfill ───────────────────────────────
+-- event-flow-simulator 는 partner_member_permissions 멤버 계정이 있는 파트너만 인증하고,
+-- partner_create_event 는 비-closed party 를 보유한 파트너만 이벤트를 만들 수 있다. 인증가능
+-- 파트너 중 party 보유 파트너가 소수면(예: 95명 중 10명) MIN_SCHEDULED_EVENTS 캡과 맞물려
+-- 이벤트 공급이 말라 user_apply 가 같은 풀을 두드리며 409 cascade 가 발생한다.
+-- 인증가능 파트너 중 비-closed party 가 하나도 없는 경우 시뮬레이터용 party 를 1개 만든다.
+-- 멱등: 이미 비-closed party 가 있으면 건너뜀.
+DO $$
+DECLARE
+  sim_partner_id uuid;
+  sim_location_id uuid;
+BEGIN
+  FOR sim_partner_id IN
+    SELECT DISTINCT pmp.partner_id
+    FROM public.partner_member_permissions pmp
+    WHERE NOT EXISTS (
+      SELECT 1 FROM public.parties pa
+      WHERE pa.partner_id = pmp.partner_id
+        AND pa.status <> 'closed'
+    )
+  LOOP
+    SELECT id INTO sim_location_id FROM public.locations
+    WHERE partner_id = sim_partner_id LIMIT 1;
+
+    INSERT INTO public.parties (partner_id, location_id, title, max_participants, status, image_urls)
+    VALUES (
+      sim_partner_id, sim_location_id, '[SIM] 시뮬레이터 파티', 20, 'active',
+      ARRAY['https://picsum.photos/seed/minglit-sim-' || replace(sim_partner_id::text, '-', '') || '/800/600']
+    );
+  END LOOP;
+END $$;

--- a/supabase/tests/database/109_seed_simulator_partner_party_backfill_test.sql
+++ b/supabase/tests/database/109_seed_simulator_partner_party_backfill_test.sql
@@ -1,0 +1,182 @@
+-- Regression test for event-flow-simulator supply:
+-- seed.dev.sql must give every authenticatable partner at least one non-closed
+-- party so partner_create_event can keep producing events.
+--
+-- Self-contained: creates its own fixtures and mirrors the Phase 11 seed block.
+BEGIN;
+
+SELECT plan(5);
+
+SELECT tests.authenticate_as_service_role();
+
+CREATE TEMP TABLE _seed_sim_backfill_state (
+  no_party_partner_id uuid,
+  closed_only_partner_id uuid,
+  active_partner_id uuid
+) ON COMMIT DROP;
+
+DO $$
+DECLARE
+  no_party_partner_id uuid;
+  closed_only_partner_id uuid;
+  active_partner_id uuid;
+  no_party_user_id uuid;
+  closed_only_user_id uuid;
+  active_user_id uuid;
+BEGIN
+  no_party_user_id := tests.create_supabase_user(
+    'seed_sim_no_party',
+    'seed_sim_no_party@pgtap.local'
+  );
+  closed_only_user_id := tests.create_supabase_user(
+    'seed_sim_closed_only',
+    'seed_sim_closed_only@pgtap.local'
+  );
+  active_user_id := tests.create_supabase_user(
+    'seed_sim_active',
+    'seed_sim_active@pgtap.local'
+  );
+
+  INSERT INTO public.partners (name)
+  VALUES ('Seed Sim No Party Partner')
+  RETURNING id INTO no_party_partner_id;
+
+  INSERT INTO public.partners (name)
+  VALUES ('Seed Sim Closed Only Partner')
+  RETURNING id INTO closed_only_partner_id;
+
+  INSERT INTO public.partners (name)
+  VALUES ('Seed Sim Active Partner')
+  RETURNING id INTO active_partner_id;
+
+  INSERT INTO public.partner_member_permissions (partner_id, user_id, role)
+  VALUES
+    (no_party_partner_id, no_party_user_id, 'owner'),
+    (closed_only_partner_id, closed_only_user_id, 'owner'),
+    (active_partner_id, active_user_id, 'owner');
+
+  INSERT INTO public.parties (partner_id, title, status, image_urls)
+  VALUES
+    (closed_only_partner_id, 'Closed Seed Sim Party', 'closed', '{}'),
+    (active_partner_id, 'Existing Active Seed Sim Party', 'active', '{}');
+
+  INSERT INTO _seed_sim_backfill_state
+  VALUES (no_party_partner_id, closed_only_partner_id, active_partner_id);
+END $$;
+
+CREATE OR REPLACE FUNCTION pg_temp._seed_simulator_party_backfill()
+RETURNS void AS $$
+DECLARE
+  sim_partner_id uuid;
+  sim_location_id uuid;
+BEGIN
+  FOR sim_partner_id IN
+    SELECT DISTINCT pmp.partner_id
+    FROM public.partner_member_permissions pmp
+    WHERE NOT EXISTS (
+      SELECT 1 FROM public.parties pa
+      WHERE pa.partner_id = pmp.partner_id
+        AND pa.status <> 'closed'
+    )
+  LOOP
+    SELECT id INTO sim_location_id FROM public.locations
+    WHERE partner_id = sim_partner_id LIMIT 1;
+
+    INSERT INTO public.parties (
+      partner_id,
+      location_id,
+      title,
+      max_participants,
+      status,
+      image_urls
+    )
+    VALUES (
+      sim_partner_id,
+      sim_location_id,
+      '[SIM] 시뮬레이터 파티',
+      20,
+      'active',
+      ARRAY[
+        'https://picsum.photos/seed/minglit-sim-' ||
+        replace(sim_partner_id::text, '-', '') ||
+        '/800/600'
+      ]
+    );
+  END LOOP;
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT is(
+  (
+    SELECT count(*)::int
+    FROM public.parties
+    WHERE partner_id IN (
+      SELECT no_party_partner_id FROM _seed_sim_backfill_state
+      UNION ALL
+      SELECT closed_only_partner_id FROM _seed_sim_backfill_state
+    )
+      AND status <> 'closed'
+  ),
+  0,
+  'fixtures start without non-closed parties for exhausted partners'
+);
+
+SELECT pg_temp._seed_simulator_party_backfill();
+
+SELECT is(
+  (
+    SELECT count(*)::int
+    FROM public.parties
+    WHERE partner_id = (
+      SELECT no_party_partner_id FROM _seed_sim_backfill_state
+    )
+      AND status <> 'closed'
+  ),
+  1,
+  'backfill creates a non-closed party when partner has no parties'
+);
+
+SELECT is(
+  (
+    SELECT count(*)::int
+    FROM public.parties
+    WHERE partner_id = (
+      SELECT closed_only_partner_id FROM _seed_sim_backfill_state
+    )
+      AND status <> 'closed'
+  ),
+  1,
+  'backfill creates a non-closed party when partner only has closed parties'
+);
+
+SELECT is(
+  (
+    SELECT count(*)::int
+    FROM public.parties
+    WHERE partner_id = (
+      SELECT active_partner_id FROM _seed_sim_backfill_state
+    )
+      AND status <> 'closed'
+  ),
+  1,
+  'backfill does not duplicate partners that already have a non-closed party'
+);
+
+SELECT is(
+  (
+    SELECT count(*)::int
+    FROM public.parties
+    WHERE title = '[SIM] 시뮬레이터 파티'
+      AND status = 'active'
+      AND partner_id IN (
+        SELECT no_party_partner_id FROM _seed_sim_backfill_state
+        UNION ALL
+        SELECT closed_only_partner_id FROM _seed_sim_backfill_state
+      )
+  ),
+  2,
+  'backfilled simulator parties use the expected active marker'
+);
+
+SELECT * FROM finish();
+ROLLBACK;

